lib/banyan_api/shop.ex

```diff
@@ -48,7 +48,8 @@ defmodule BanyanAPI.Shop do
     } =
       Map.merge(
         @default_settings,
-        settings
+        settings,
```

Comment:
This is the `settings` JSON blob straight out of the db, yeah? How does it end up with null values?

Reply:
I'm just wondering if it should maybe be the responsibility of whatever module calls this function to ensure it's passing in clean data.

Reply:
I agree they shouldn't be null as the only way that can happen is if Shopify passes us bad datas somehow, and going into the shop and resaving their settings fixed the issue (the null value was replaced with an empty string). 

This could be a one off edge case, hard to tell.

When it comes to ensuring clean data its easier to deal with it here than find everywhere its called to sanitize the data, or everwhere the DB is updated.

Reply:
Yeah. If we're going to deal with this here, I'd maybe prefer something like:
```
Map.merge(
     @default_settings,
     remove_nil_values(settings)
)
```
as this is a bit difficult to parse on first glance.

Reply:
thats valid

Reply:
I've implemented this solution, I agree its more readable